impl/src/main/java/org/jboss/weld/bean/DisposalMethod.java

```diff
@@ -123,7 +123,7 @@ private void checkDisposalMethod(EnhancedAnnotatedMethod<T, ? super X> enhancedA
         }
         if (declaringBean instanceof SessionBean<?>) {
             SessionBean<?> sessionBean = (SessionBean<?>) declaringBean;
-            Set<MethodSignature> businessMethodSignatures = sessionBean.getBusinessMethodSignatures();
+            Set<MethodSignature> businessMethodSignatures = sessionBean.getLocalBusinessMethodSignatures();
```

Comment:
Good point. I think it could be nowadays however the question is whether it makes sense when you call this method "remotely" - which means you call it by name and do not really on CDI to produce the instance for you. I think it could be worth of spec issue.